### PR TITLE
update travis config based on https://config.travis-ci.com/explore

### PR DESCRIPTION
**Pull request by amaliujia**
Travis-ci now has an online tool to validate travis config: https://config.travis-ci.com/explore. 

I updated travis config based on the suggestions from the tool. E.g. fix warnings.

### CODE REVIEW

**Comment by amaliujia** [comment]
@aaltay @boyuanzz

File: .travis.yml [hunk 1]
```diff
@@ -16,7 +15,7 @@ env:
       - BUILD_DEPENDS="Cython"
       - TEST_DEPENDS=
 
-matrix:
+jobs:
   exclude:
     # Exclude the default Python 3.5 build
     - python: 3.5
```
Comment (amaliujia) [inline .travis.yml]: The python version is 3.5 above but 3.5 is specified as "exclude" here. Is it the reason that making the build skipped?

Comment (amaliujia) [reply .travis.yml]: Actually this link might gives the real reason: https://docs.travis-ci.com/user/build-matrix/#explicitly-included-jobs-with-only-one-element-in-the-build-matrix

File: .travis.yml [hunk 2]
```diff
@@ -96,9 +95,9 @@ deploy:
   access_key_id: ${ACCESS_KEY_ID}
   secret_access_key: ${SECRET_ACCESS_KEY}
   bucket: "beam-wheels-staging"
-  skip_cleanup: true
```
Comment (aaltay) [inline .travis.yml:99]: why was this true?

File: .travis.yml (hunk 1 above)
Comment (aaltay) [reply .travis.yml]: I am not sure what this does.

**Review by aaltay — APPROVED** [review]
Go for it if it works. I am not sure how to verify travis config changes short of running and seeing it produces the results.

File: .travis.yml (hunk 2 above)
Comment (amaliujia) [reply .travis.yml:99]: `skip_cleanup` is deprecated and it's replacement is `cleanup`

I should have provided a list of warning messages from https://config.travis-ci.com/explore

**Comment by amaliujia** [comment]
This change has fixed the "does not trigger build" problem.

**Comment by aaltay** [comment]
> This change has fixed the "does not trigger build" problem.

Thank you!

**Pull request merged**